uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/TOILRequestTest.php

```diff
@@ -94,6 +94,18 @@ public function testCreateTOILRequestWithPastDatesAndAbsenceTypeAllowsIt() {
 
     $fromDate = new DateTime('2016-11-14');
     $toDate = new DateTime('2016-11-18');
+
+    // If this throws an exception, The test will fail
+    TOILRequest::validateParams([
+      'type_id' => $absenceType->id,
+      'contact_id' => 1,
+      'status_id' => 1,
+      'from_date' => $fromDate->format('YmdHis'),
+      'to_date' => $toDate->format('YmdHis'),
+      'toil_to_accrue' => $this->toilAmounts['2 Days']['value'],
+      'duration' => 120
+    ]);
+
```

Comment:
I don't understand the reason for this.

It would be better to have a test to make sure the validateParams is called inside create when $validate = true. Something like this would be enough, I think:
```php
/**
* @expectedException CRM_HRLeaveAndAbsences_Exception_InvalidTOILRequestException
* @expectedExceptionMessage You cannot request TOIL for past days
*/
public function testValidateParamsIsCalledOnCreate() {
  // code to request toil for past days
}
```

You don't need to test all possible exceptions. Only one is enough to make sure validaParams has been called